pkgs/development/python-modules/keyutils/default.nix

```
@@ -16,7 +16,12 @@ buildPythonPackage rec {
     substituteInPlace setup.py --replace '"pytest-runner"' ""
   '';
 
+  postConfigure = ''
+    ${cython}/bin/cython keyutils/_keyutils.pyx
```

Why not
```suggestion
    cython keyutils/_keyutils.pyx
```
?
Anyway, the correct to do what you're doing right now would be
```suggestion
    ${lib.getExe' cython "cython"} keyutils/_keyutils.pyx
```

---

Using the bare path fails:

```
❯ nix build .#python311Packages.keyutils
warning: Git tree '/home/chrism/src/nixpkgs' is dirty
error: builder for '/nix/store/5r6jhm1h9mkpbji4csip8bf5j99h77pg-python3.11-keyutils-0.6.drv' failed with exit code 127;
       last 10 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/dxc4n5ln8sm8rm6hpkjxgzfd3c6fi0wz-source
       > source root is source
       > setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
       > patching sources
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > no configure script, doing nothing
       > /nix/store/j4w7yj8c4z7zg5y6bhhypnl456dwmrc5-stdenv-linux/setup: line 127: cython: command not found
       > /nix/store/j4w7yj8c4z7zg5y6bhhypnl456dwmrc5-stdenv-linux/setup: line 144: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/5r6jhm1h9mkpbji4csip8bf5j99h77pg-python3.11-keyutils-0.6.drv'.
```

I'm updating to use getExe'.

---

It fails because you misspelled `nativeBuildInputs`.

---

Corrected and build works now with the bare invocation.

---

> Is there any tooling (linter, etc) that I can use to check for these kinds of errors in the future aside from just knowing that it will available in the build environment PATH?

Not that I know of sadly.

---

:+1: good to know in the meantime - thanks  for the feedback!